File: Data Structures & Algorithms/sum-of-two-integers/submission-0.py
class Solution:
    def getSum(self, a: int, b: int) -> int:
        total = 0
        carry = 0
        for i in range(32):
            a_bit = (a >> i) & 1
            b_bit = (b >> i) & 1

            curr = a_bit ^ b_bit ^ carry
            carry = 1 if a_bit + b_bit + carry >= 2 else 0
            total |= (curr << i)

        if total > 0x7FFFFFFF:
            total = ~(total ^ 0xFFFFFFFF)

        return total

File: Data Structures & Algorithms/climbing-stairs/submission-0.java
class Solution {
    public int climbStairs(int n) {
       int curr = 0;
       int dp1 = 1;
       int dp2 = 0;
       for (int i = n; i >= 1; i--) {
            curr = dp1 + dp2;
            dp2 = dp1;
            dp1 = curr;
            curr = 0;
       }
       return dp1;
    }
}
